src/window.mm

```diff
@@ -148,6 +148,40 @@ - (void)notified:(const std::string &)note withData:(const msgpack::object &)upd
         AppDelegate *app = (AppDelegate *)[NSApp delegate];
         [app newWindow];
     }
+    else if (note == "neovim.app.openFile") {
+        NSArray *files = [mMainView showFileOpenDialog];
+        if (files != nil) {
+            std::vector<char *> args;
+
+            /* open in tabs if more than one file was selected */
+            if ([files count] > 1)
+                args.push_back(const_cast<char *>("-p"));
+
+            for (NSURL *url in files) 
+                args.push_back(const_cast<char *>([[url path] UTF8String]));
+
+            AppDelegate *app = (AppDelegate *)[NSApp delegate];
+            [app newWindowWithArgs:args];
```

Comment:
Just checked MacVim again and it looks like it's a setting in preferences. (Called "open files from applications" but applies to file menu too)